config/kernel-fpu.m4

```diff
@@ -12,25 +12,49 @@ dnl # Pre-4.2:	Use kernel_fpu_{begin,end}()
 dnl #		HAVE_KERNEL_FPU & KERNEL_EXPORTS_X86_FPU
 dnl #
 AC_DEFUN([ZFS_AC_KERNEL_FPU], [
-	AC_MSG_CHECKING([which kernel_fpu function to use])
+	AC_MSG_CHECKING([which kernel_fpu header to use])
 	ZFS_LINUX_TRY_COMPILE([
+		#include <linux/module.h>
+		#include <asm/fpu/api.h>
+	],[
+	],[
+		AC_DEFINE(HAVE_KERNEL_FPU_API_HEADER, 1, [kernel has asm/fpu/api.h])
+		AC_MSG_RESULT(asm/fpu/api.h)
+	],[
+		AC_MSG_RESULT(i387.h & xcr.h)
+	])
+
+	AC_MSG_CHECKING([which kernel_fpu function to use])
+	ZFS_LINUX_TRY_COMPILE_SYMBOL([
+		#include <linux/module.h>
+		#ifdef HAVE_KERNEL_FPU_API_HEADER
+		#include <asm/fpu/api.h>
+		#else
 		#include <asm/i387.h>
 		#include <asm/xcr.h>
+		#endif
 	],[
+		MODULE_LICENSE(ZFS_META_LICENSE);
```

Comment:
This should be in the previous block, outside the `main()` function.  It may also need to be quoted, take a look at `ZFS_AC_KERNEL_BLK_QUEUE_FLUSH` for an example.